### DISCUSSION TITLE
org-roam-bibtex-mode did not change the behavior of ivy-bibtex/helm-bibtex

### DISCUSSION THREAD

**Discussion by wztdream** [org-roam/org-roam-bibtex#249] [Q&A]
Hi,
I read the code of `org-roam-bibtex-mode`, but it seems it did not change the behavior of `ivy-bibtex` or `helm-bibtex`. Is it intended or not?
https://github.com/org-roam/org-roam-bibtex/blob/201262a839db20af2a49165a80f85f82dad159d1/org-roam-bibtex.el#L1113-L1163
In current version the behavior change are only happens when called `org-insert-link`
https://github.com/org-roam/org-roam-bibtex/blob/201262a839db20af2a49165a80f85f82dad159d1/orb-ivy.el#L85-L88

Personally I prefer change the `ivy-bibtex` or `helm-bibtex` behavior in `org-roam-bibtex-mode`, because this is a clear mental mode that change all bibtex related function to org-roam style.

**Comment by myshevchuk** [comment]
Hi,

I don't think I understand what change in behaviour of Helm/Ivy-bibtex you mean. Could you please elaborate, with an example?

**Reply by wztdream** [reply]
> Hi,
> 
> I don't think I understand what change in behaviour of Helm/Ivy-bibtex you mean. Could you please elaborate, with an example?

Now I under stand the logic, `org-roam-bibtex-mode` change the value of `bibtex-completion-edit-notes-function`, so changed the behavior of ivy/helm-bibtex.
